### PR TITLE
Update DBotTrainClustering docker

### PR DESCRIPTION
**Pull request by jlevypaloalto**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-11147)

## Description
Update DBotTrainClustering docker.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Base/Scripts/DBotTrainClustering/DBotTrainClustering.py [hunk 1]
```diff
@@ -153,8 +150,10 @@ def reduce_dimension(self, dimension=2):
         :return:
         """
         if not self.TSNE_:
-            tsne = TSNE(n_jobs=-1, n_components=dimension, learning_rate=1000)
-            self.data_2d = tsne.fit_transform(pd.DataFrame(self.centers).T)
+            samples = pd.DataFrame(self.centers).T
+            perplexity = float(min(30, samples.shape[0] - 1))
+            tsne = TSNE(perplexity=perplexity, n_jobs=-1, n_components=dimension, learning_rate=1000)
```
Comment (JudahSchwartz) [inline Packs/Base/Scripts/DBotTrainClustering/DBotTrainClustering.py:155]: lets check out what this change entails, what is perplexity and what is happening/whats the impact by lowering it?

File: Packs/Base/Scripts/DBotTrainClustering/DBotTrainClustering.py [hunk 2]
```diff
@@ -637,46 +631,48 @@ def create_summary(model_processed: Type[PostProcessing], fields_for_clustering:
...
+            'incidents_ids': clst['incidents_ids']
+        }
+        for clst in json.loads(output_clustering)['data']
+    ]
     return_entry = {
         "Type": entryTypes["note"],
         "ContentsFormat": formats['json'],
         "Contents": output_clustering,
-        "EntryContext": {'DBotTrainClustering': output_clustering},
+        "EntryContext": {'DBotTrainClustering': outputs},
```
Comment (JudahSchwartz) [inline Packs/Base/Scripts/DBotTrainClustering/DBotTrainClustering.py:668]: does this break the context outputs? if yes, lets add all outputs now to the yml and release with breaking changes. 
Do we need to have breaking changes?

File: Packs/Base/Scripts/DBotTrainClustering/DBotTrainClustering.py [hunk 3]
```diff
@@ -701,41 +697,30 @@ def fill_nested_fields(incidents_df: pd.DataFrame, incidents: List, *list_of_fie
...
                     if not keep_unique_value:
-                        value_list = [' '.join(  # type: ignore
-                            set(
-                                list(
-                                    filter(lambda x: x not in ['None', None, 'N/A'], x)
-                                )
-                            )
-                        )
-                            for x in value_list]
+                        value_list = [' '.join(set(filter(lambda x: x not in ['None', None, 'N/A'], x))) for x in value_list]
```
Comment (JudahSchwartz) [inline Packs/Base/Scripts/DBotTrainClustering/DBotTrainClustering.py:702]: use if not a filter
this syntax -> [x for x in x if x]

File: Packs/Base/Scripts/DBotTrainClustering/DBotTrainClustering.py [hunk 4]
```diff
@@ -754,54 +739,39 @@ def remove_not_valid_field(fields_for_clustering: List[str], incidents_df: pd.Da
...
-    Load model from base64 model
-    :param model_base64: string base64 model
-    :return: PostProcessing model
-    """
-    try:
-        model = pickle.loads(base64.b64decode(model_base64))  # guardrails-disable-line
-        return model
-    except pickle.UnpicklingError:
-        return_error("Model exist but cannot be loaded")
+        return None
```
Comment (JudahSchwartz) [inline Packs/Base/Scripts/DBotTrainClustering/DBotTrainClustering.py:769]: make sure well tested manualky and w unit tests

File: Packs/Base/Scripts/DBotTrainClustering/README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+Train clustering model on any incident type.
```
Comment (JudahSchwartz) [inline Packs/Base/Scripts/DBotTrainClustering/README.md:1]: give a bit more details on the usecase

**Review by JudahSchwartz — COMMENTED** [review]
fantastic! great work!

File: Packs/Base/Scripts/DBotTrainClustering/README.md (hunk 5 above)
Comment (JudahSchwartz) [reply Packs/Base/Scripts/DBotTrainClustering/README.md]: still relevant

File: Packs/Base/Scripts/DBotTrainClustering/README.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+## Script Data
+
+---
+
+| **Name** | **Description** |
+| --- | --- |
+| Script Type | python3 |
+| Tags | ml |
+| Cortex XSOAR Version | 6.2.0 |
```
Comment (JudahSchwartz) [inline Packs/Base/Scripts/DBotTrainClustering/README.md:13]: this is an input? what does this mean xsoar version

File: Packs/Base/Scripts/DBotTrainClustering/README.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+## Inputs
+
+---
+
+| **Argument Name** | **Description** |
+| --- | --- |
```
Comment (JudahSchwartz) [inline Packs/Base/Scripts/DBotTrainClustering/README.md:20]: tag for doc review

**Review by JudahSchwartz — APPROVED** [review]
please tag doc review, and try to hit resolve when resolving a comment that way i can see what still needs to be looked at

File: Packs/Base/Scripts/DBotTrainClustering/README.md (hunk 6 above)
Comment (jlevypaloalto) [reply Packs/Base/Scripts/DBotTrainClustering/README.md:13]: it's auto-generated

File: Packs/Base/ReleaseNotes/1_34_31.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+
+#### Scripts
+
+##### DBotTrainClustering
+
+- Updated the Docker image to: *demisto/mlclustering:1.0.0.105775*.
```
Comment (ShirleyDenkberg) [inline Packs/Base/ReleaseNotes/1_34_31.md:6]:
```suggestion
Updated the Docker image to: *demisto/mlclustering:1.0.0.105775*.
```

File: Packs/Base/Scripts/DBotTrainClustering/README.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+| fieldsForClustering | Comma-separated list of incident fields to take into account when training the clustering. |
+| fieldForClusterName | Incident field that represents the family name for each cluster created. The model determines how many incidents in the cluster have the same value in the fieldForClusterName field. The largest numbers of incidents with the same value, determine the cluster name. |
```
Comment (ShirleyDenkberg) [inline Packs/Base/Scripts/DBotTrainClustering/README.md:22]:
```suggestion
| fieldForClusterName | Incident field that represents the family name for each cluster created. The model determines how many incidents in the cluster have the same value in the fieldForClusterName field. The largest numbers of incidents with the same value determine the cluster name. |
```

**Comment by ShirleyDenkberg** [comment]
@JudahSchwartz @sapirshuker Doc review completed.

**Pull request merged**